doc/sourmash-internals.md

```diff
@@ -0,0 +1,638 @@
+keeping all hashes below `2**64 / scaled` (for FracMinHash sketches).
+
+The default MinHash sketches use parameters so that they are
+compatible with mash sketches.
+
+See [utils/compute-dna-mh-another-way.py](https://github.com/sourmash-bio/sourmash/blob/latest/utils/compute-dna-mh-another-way.py) for details on how k-mers are
+hashed.
+
+Note that if hashes are produced some other way (with a different hash
+function) or from some source other than DNA, sourmash can still work
+with them; only `sourmash sketch` does input checking.
+
+### Compatibility checking
+
+The point of the signatures and sketches is to enable certain kinds of
+rapid comparisons - Jaccard similarity and overlap analysis,
+specifically. However, these comparisons can only be done between
+compatible sketches.
+
+Here, "compatible" means -
+* the same MurmurHash seed (default 42);
+* the same k-mer size/ksize (see k-mer sizes, below);
+* the same molecule type (see molecule types, below);
+* the same `num` or `scaled` (although see [this downsampling discussion](api-example.md#downsampling-signatures), and the next two sections);
+
+sourmash uses selectors (`Index.select(...)`) to select sketches with
+compatible ksizes, molecule types, and sketch types.
+
+### Scaled (FracMinHash) sketches support similarity and containment
+
+Per our discussion in [Irber et al., 2022](https://www.biorxiv.org/content/10.1101/2022.01.11.475838v2), FracMinHash sketches can always be compared
+by downsampling to the max of the two scaled values.  (This is not always
+true of sketches stored in indexed collections, e.g. SBTs; see [sourmash#1799](https://github.com/sourmash-bio/sourmash/issues/1799).)
+
+In practice, sourmash does all necessary downsampling dynamically, but
+returns the original sketches. This means that (for example) you can
+do a low-resolution/high-scaled large scale search by specifying a
+high scaled value, and then do a higher resolution comparison with
+only the highly similar matches the results to do a more refined (see
+below, Speeding up `gather` and `search`.)
+
+### Num (MinHash) sketches support Jaccard similarity
+
+"Regular" MinHash (or "num MinHash") sketches are implemented the same
+way as in mash.  However, they are less well supported in sourmash,
+because they don't offer the same opportunities for metagenome
+analysis.  (See also [sourmash#1354](https://github.com/sourmash-bio/sourmash/issues/1354).)
+
+Num MinHash sketches can always be compared by downsampling to a
+common `num` value. This may need to be done manually using `sourmash
+sig downsample`, however.
+
+## K-mer sizes
+
+There is no explicit restriction on k-mer sizes built into sourmash.
+
+For highly specific genome and metagenome comparisons, we typically
+use k=21, k=31, or k=51. For a longer discussion, see [Assembly Free Analysis with K-mers](https://github.com/mblstamps/stamps2022/blob/main/kmers_and_sourmash/2022-stamps-assembly-free%20class.pdf) from STAMPS 2022.
```

Comment:
```suggestion
use k=21, k=31, or k=51. For a longer discussion, see [Assembly Free Analysis with K-mers](https://github.com/mblstamps/stamps2022/blob/main/kmers_and_sourmash/2022-stamps-assembly-free%20class.pdf) from STAMPS 2022 and a more general overview at [Using sourmash:a practical guide](https://sourmash.readthedocs.io/en/latest/using-sourmash-a-guide.html). 
```

Reply:
thanks! you can link directly within the docs by using Sphinx internal references - here, the reference would be `using-sourmash-a-guide.md`. I'm updating it in a separate commit at the moment ;)

Reply:
in fbbcee56

Reply:
Interesting. Thanks for the info!